base/card-api.gts

```diff
@@ -990,6 +975,73 @@ export class Box<T> {
     return newChildren;
   }
 
+  // trying to determine the card from the outside of the box destabilizes otherwise stable
+  // arrays due to glimmer invalidation in our field getters. I'm breaking the rule where
+  // boxes aren't cognizant of cards to work around this situation
+  get card() {
+    if (this.state.type === 'root') {
+      throw new Error('tried to call card() on root box');
+    }
+    return typeof this.value === 'object' && this.value != null && isBaseCard in this.value
+      ? Reflect.getPrototypeOf(this.value as unknown as Card)!.constructor as typeof Card 
+      : undefined;
+  }
```

Comment:
this was an important finding that allowed me to determine the field card to render with without destabilizing array models